app/assets/javascripts/student_profile/page_container.js

```diff
@@ -94,10 +94,17 @@
     },
 
     onClickSaveService: function(serviceParams) {
-      this.setState({ requests: merge(this.state.requests, { saveService: 'pending' }) });
-      this.api.saveService(this.state.student.id, serviceParams)
-        .done(this.onSaveServiceDone)
-        .fail(this.onSaveServiceFail);
+        // Very quick name validation, just check for a comma
+        if ((/\w+, \w+/.test(serviceParams.providedByEducatorName)) ||
```

Comment:
You're checking against null to ensure that it's not empty, right? Would it be simpler to store whatever the user types in `onProvidedByEducatorTyping` and just check against empty here?

I think then you could even drop the second check entirely, and let the regex take care of checking for empty.